staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -573,3 +591,34 @@ func SearchMountPoints(hostSource, mountInfoPath string) ([]string, error) {
 
 	return refs, nil
 }
+
+// tryUnmount calls plain "umount" and waits for unmountTimeout for it to finish.
+func tryUnmount(path string, unmountTimeout time.Duration) error {
+	klog.V(4).Infof("Unmounting %s", path)
+	ctx, cancel := context.WithTimeout(context.Background(), unmountTimeout)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "umount", path)
+	out, cmderr := cmd.CombinedOutput()
+
+	// CombinedOutput() does not return DeadlineExceeded, make sure it's
+	// propagated on timeout.
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	if cmderr != nil {
+		return fmt.Errorf("unmount failed: %v\nUnmounting arguments: %s\nOutput: %s", cmderr, path, string(out))
+	}
+	return nil
+}
+
+func forceUmount(path string) error {
+	cmd := exec.Command("umount", "-f", path)
```

Comment:
so should we consider to have a timeout even for force unmount?

Reply:
the same question for normal Unmount function

Reply:
> the same question for normal Unmount function.

For normal unmount it is bit problematic to change behaviour for all existing drivers. As we know that some storage types flush pending writes when a volume is unmounted - and hence unmount time could vary and it is not safe to kill unmount process when this happens. So I think we never want to use a universal timeout for normal unmount function for all drivers. 

> so should we consider to have a timeout even for force unmount?

force works best for unreachable NFS system. It is documented directly in the man page. But it is not guaranteed to not hang  and as far as I know if it hangs in other cases, it could be hanging for a legitimate reason (such as volume being used or something else) and in those cases using a timeout will basically kill the unmount process and it could be problematic. So, I think I would not want to use a timeout for force unmount. It could result in a leaked goroutine but that is okay, we always leaked goroutines in this code paths.